op-challenger/fault/responder_test.go

```diff
@@ -0,0 +1,42 @@
+package fault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	// "github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/require"
+)
+
+// type mockTxData struct {
+// 	data []byte
+// }
+//
+// func (m *mockTxData) Bytes() []byte {
+// 	return m.data
+// }
+//
+// func (m *mockTxData) Destination() *common.Address {
+// 	return &common.Address{}
+// }
+//
+// type mockTxQueue struct{}
+//
+// func (m *mockTxQueue) Send(txdata TxData, candidate txmgr.TxCandidate, receiptsCh chan txmgr.TxReceipt[TxData]) {
+// 	receiptsCh <- txmgr.TxReceipt[TxData]{}
+// }
+
+// TestResponder_sendTransaction_NoData tests the sendTransaction method
+// with no transaction data provided by the txdata.
+func TestResponder_sendTransaction_NoData(t *testing.T) {
+	t.Skip("Responder not tested yet")
+
+	receiptChan := make(chan txmgr.TxReceipt[TxData])
+	err := fmt.Errorf("error")
+	// go func() {
+	// 	err = sendTransaction(&mockTxData{}, &mockTxQueue{}, receiptChan)
+	// }()
+	<-receiptChan
+	require.NoError(t, err)
+}
```

Comment:
We need to be writing tests for all the parts as we go and makings sure the design is actually testable.  Needing to spawn a thread in a unit test is a sign we haven't got that design right yet.

Reply:
Agreed - have been for most. Some other game solver components haven't been tested and are merged since we are still trying to get the design right i believe - architecture hasn't been sussed out yet.

That said, I can write more tests here, they will likely be deleted or clobbered though if the architecture changes so it'd take quite a bit of my time

Reply:
Ultimately you'll go faster if you write tests as you go. Yes you might rework some of them but if you write tests that operate on the public interfaces rather than depending on internal details they won't be brittle and can handle having things change later without needing major rework.  Having tests will also push you to use a better design because code is only testable when you have appropriate coupling and a whole bunch of other key properties of high quality code.  ie writing tests is actually a key part of the design process, not something to tack on the end.

Reply:
Most of the changes we've made have been tested & then only time we've updated the tests is when we change the types of stuff. The couple things that don't have unit tests probably should, but have been tested manually / are exercised through the examples.

For now I think we probably don't need concurrency in the transaction sending which will make testing much easier.